DFC.Digital/DFC.Digital.Web.Sitefinity.Core/CompositeUI/ServiceBusMessageProcessor.cs

```diff
@@ -35,14 +35,37 @@ public async Task SendJobProfileMessage(JobProfileMessage jpData, string content
 
             // Message that send to the queue
             var message = new Message(Encoding.UTF8.GetBytes(jsonData));
-            Console.WriteLine($"Sending message to queue: {jsonData}");
             message.ContentType = "application/json";
             message.Label = jpData.Title;
             message.UserProperties.Add("Id", jpData.JobProfileId);
             message.UserProperties.Add("EventType", actionType);
             message.UserProperties.Add("CType", contentType);
             await topicClient.SendAsync(message);
             applicationLogger.Info($"{Constants.ServiceStatusPassedLogMessage} {jpData.JobProfileId.ToString()}");
+            await topicClient.CloseAsync();
+        }
+
+        public async Task SendOtherRelatedTypeMessages(IEnumerable<RelatedContentItem> relatedContentItems, string contentType, string actionType)
+        {
+            var connectionStringServiceBus = configurationProvider.GetConfig<string>("DFC.Digital.ServiceBus.ConnectionString");
+            var topicName = configurationProvider.GetConfig<string>("DFC.Digital.ServiceBus.TopicName");
+            var topicClient = new TopicClient(connectionStringServiceBus, topicName);
+
+            foreach (var relatedContentItem in relatedContentItems)
+            {
+                // Send Messages
+                var jsonData = JsonConvert.SerializeObject(relatedContentItem);
+
+                // Message that send to the queue
+                var message = new Message(Encoding.UTF8.GetBytes(jsonData));
+                message.ContentType = "application/json";
+                message.Label = relatedContentItem.Title;
+                message.UserProperties.Add("Id", $"{relatedContentItem.JobProfileId}--{relatedContentItem.Id}");
+                message.UserProperties.Add("EventType", actionType);
+                message.UserProperties.Add("CType", contentType);
+                await topicClient.SendAsync(message);
+                applicationLogger.Info($"{Constants.ServiceStatusPassedLogMessage} {relatedContentItem.JobProfileId}--{relatedContentItem.Id}");
+            }
 
             await topicClient.CloseAsync();
```

Comment:
You may have to close in finally block to avoid any connection leak on failures

Reply:
Added Finally block